### ISSUE TITLE
Add `UpgradeSequencePath` to `24-host`

### ISSUE THREAD

**Issue by damiannolan** [cosmos/ibc-go#1600]
## Summary

> The upgrade sequence path is a public path that stores the current sequence of the upgrade attempt. The sequence will increment with each attempted upgrade on the given channel. The sequence will be used to ensure that different error receipts referring to different upgrade attempts do not interfere with each other.

Add `UpgradeSequencePath` to `24-host` with associated key func.

- `UpgradeSequencePath`
- `UpgradeSequenceKey`

Ref: [24-host/keys.go](https://github.com/cosmos/ibc-go/blob/main/modules/core/24-host/keys.go)

```typescript
function upgradeSequencePath(portIdentifier: Identifier, channelIdentifier: Identifier) Path {
    return "channelUpgrade/ports/{portIdentifier}/channelIdentifier/{channelIdentifier}/upgradeSequence"
}
```

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by damiannolan** [comment]
Review spec for any updates to key paths

**Comment by chatton** [comment]
closed by #1823
